enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1179,14 +1190,56 @@ func (dc *writeCloser) Write(p []byte) (int, error) {
 	return n, nil
 }
 
-// TODO(Maggie): Compress all data above a given size before writing it
+// zstdCompressor compresses bytes before writing them to the nested writer
+type zstdCompressor struct {
```

Comment:
wdyt about having a New method for this struct that accepts a bufferPool and ioutil.CustomCommitWriteCloser and calls bufferPool.Get()? That way the bufferpool lifecycle is fully internal to this struct instead of being half the callers responsibility and half this code's responsibility?

Reply:
Done